### ISSUE TITLE
Push to library requires extra '/'

### ISSUE THREAD

**Issue by dtrudg** [sylabs/singularity#691]
**Version of Singularity**

3.9.7

**Describe the bug**

```
10:00 AM $ singularity push -U ubuntu_latest.sif library://dtrudg-sylabs-2/test/ubuntu:test
WARNING: Skipping container verification
FATAL:   Unable to push image to library: malformed image path: dtrudg-sylabs-2test/ubuntu
dtrudg@p700:~
10:00 AM $ singularity push -U ubuntu_latest.sif library://dtrudg-sylabs-2//test/ubuntu:test
WARNING: Skipping container verification
11.0MiB / 35.5MiB [=======================>------------------------------------------------------] 31 % 998.5 KiB/s 25s
```

Occurs on 3.9.7 but not 3.9.6

Relevant change is scs-library-client update from 1.2.1 to 1.2.2

Debug output:

```
$ singularity -d push -U ubuntu_latest.sif library://dtrudg-sylabs-2/test/ubuntu:test
DEBUG   [U=1000,P=130103]  persistentPreRun()            Singularity version: 3.9.7
DEBUG   [U=1000,P=130103]  persistentPreRun()            Parsing configuration file /usr/local/etc/singularity/singularity.conf
DEBUG   [U=1000,P=130103]  handleConfDir()               /home/dtrudg/.singularity already exists. Not creating.
DEBUG   [U=1000,P=130103]  handleRemoteConf()            Ensuring file permission of 0600 on /home/dtrudg/.singularity/remote.yaml
WARNING [U=1000,P=130103]  LibraryPush()                 Skipping container verification
FATAL   [U=1000,P=130103]  func52()                      Unable to push image to library: malformed image path: dtrudg-sylabs-2test/ubuntu
```

**Comment by tri-adam** [comment]
I believe this is an interaction between [scs-library-client@v1.2.2](https://github.com/sylabs/scs-library-client/commit/41a19db8d156d6f8a95fc69a5932e9597d0adaf4) and this bit of the Singularity code base:

https://github.com/sylabs/singularity/blob/a86fc567fad71f6e0b64e48622899cd021658dfe/internal/app/singularity/push.go#L140

As of v1.2.2, `r.Path` no longer starts with a leading `/`. Prior to v1.2.2, _some_ ref formats would result in a leading `/`, while others would not.

Given the subtle logic required to massage the various Library ref formats, I propose adding functionality to [scs-library-client](https://github.com/sylabs/scs-library-client) to handle this. Draft up here: https://github.com/sylabs/scs-library-client/pull/139.

**Comment by tri-adam** [comment]
The changes in https://github.com/sylabs/scs-library-client/pull/139 were released in https://github.com/sylabs/scs-library-client/releases/tag/v1.3.0 and the relevant Singularity code adapted in #692 and #693. I believe this addresses this issue. Apologies 🇨🇦 for the bug!

**Comment by tri-adam** [comment]
Fixed in [v3.9.8](https://github.com/sylabs/singularity/releases/tag/v3.9.8).
